src/app/AttributeValueEncoder.h

```diff
@@ -226,7 +248,7 @@ class AttributeValueEncoder
         }
 
         mCurrentEncodingListIndex++;
-        mEncodeState.mCurrentEncodingListIndex++;
+        mEncodeState.SetCurrentEncodingListIndex(mCurrentEncodingListIndex);
```

Comment:
Here I believe the previous intent was to increase both and they seem to be kept in sync since line 220 seems to increment until they are equal.

The exact equivalent is `mEncodeState.SetCurrentEncodingListIndex(mEncodeState.CurrentEncodingListIndex() + 1)` however current "they are equal" seems clearer.

Unit and integration tests should detect if something is off - we do have tests for chunking.